crates/ruff_linter/src/rules/flake8_async/rules/cancel_scope_no_checkpoint.rs

```diff
@@ -69,12 +69,31 @@ pub(crate) fn cancel_scope_no_checkpoint(
         return;
     }
 
+    // If the body contains an `await` statement, the context manager is used correctly.
     let mut visitor = AwaitVisitor::default();
     visitor.visit_body(&with_stmt.body);
     if visitor.seen_await {
         return;
     }
 
+    // If there's an `asyncio.TaskGroup()` context manager alongside the timeout, it's fine, as in:
+    // ```python
+    // async with asyncio.timeout(2.0), asyncio.TaskGroup():
+    //     ...
+    // ```
+    if with_items.iter().any(|item| {
+        item.context_expr.as_call_expr().is_some_and(|call| {
+            checker
+                .semantic()
+                .resolve_qualified_name(call.func.as_ref())
+                .is_some_and(|qualified_name| {
+                    matches!(qualified_name.segments(), ["asyncio", "TaskGroup"])
```

Comment:
Shouldn't it hold for any async context manager? I don't think there is something unique about `asyncio.TaskGroup`.

If the `async with` is [explicitly nested](https://play.ruff.rs/351bf975-d24e-4826-a07a-9587670770db) then ruff doesn't trigger the error. So perhaps the fix should be to make

```py
async with asyncio.timeout(), foo():
    ...
```

be treated the same as

```py
async with asyncio.timeout():
    async with foo():
        ...
```

According to the [Language Reference](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement) they are semantically equivalent (this is for `with`, the [`async with` ref](https://docs.python.org/3/reference/compound_stmts.html#async-with) seems to have an omission here):

![image](https://github.com/user-attachments/assets/d9791c90-fa0a-43e9-a621-7c025782618e)


I also checked `async for` and it's detected correctly.

Reply:
Interested in changing it? :) We might need to accommodate cases like `async with asyncio.timeout(), asyncio.timeout()`, I guess those should still trigger even if weird.